src/current/v24.3/ldap-authentication.md

```diff
@@ -0,0 +1,149 @@
+---
+title: Configure LDAP Authentication
+summary: Learn how to configure CockroachDB to use LDAP for user authentication with directory services like Active Directory.
+toc: true
+---
+
+CockroachDB supports authentication and authorization using LDAP (Lightweight Directory Access Protocol) with services such as Active Directory. This allows you to integrate your existing directory service with CockroachDB for centralized user management and access control.
+
+This page describes how to configure CockroachDB user authentication using LDAP. You can also configure user authorization (role-based access control) with LDAP, to set users' CockroachDB roles based on group memberships in the directory, but this is not required.
+
+## Overview
+
+LDAP authentication in CockroachDB uses a 'search and bind' approach:
+
+1. CockroachDB first connects to the LDAP server using a service account (bind DN).
```

Comment:
```suggestion
1. The cluster connects to the LDAP server using a service account (bind DN).
```
I'm confused by the parenthetical. Are we saying that the LDAP source must have its bind DN configured in some specific way to point to a cloud infrastructure service account?

Reply:
This is correct, but is there a need to get into more specifics? Actually we have 1 connection per CockroachDB cluster node, so if there are n nodes in the cluster, we consume n ldap connections on LDAP AD service. Also to note these connections are established only if there is a request to a particular node for connecting via LDAP and are not closed unless AD server closes it.

Reply:
I'm trying to get really clear about it because the bind DN effectively sets the search scope for the LDAP search, right? Thanks for the extra details.

Reply:
> bind DN effectively sets the search scope for the LDAP search

Yes, in 1 authN attempt, we first bind as bindDN, perform search for user based on HBA attr and `sAMAccountName` provided as sql username and get user DN.
We then use the userDN over the same connection to bind to AD to see if creds were correct.
Post that we again rebind over same connection as bindDN and search for the groups the user is assigned to.
All of this is for a single auth attempt to single CRDB cluster node.

Reply:
This makes sense to me, thank you. There are several actors here so I think it's important to talk about the cluster, the directory service, the authenticating user very specifically. It would be easy to get lost in who is doing what.